fasm2bels/models/clb_models.py

```diff
@@ -1786,11 +1786,12 @@ def connect_ce_sr(bel, ce, sr):
                                      bel.bel, 'O{}'.format(idx))
 
             co_pin = 'CO[{}]'.format(idx)
-            if idx == 3:
-                site.add_source(bel, co_pin, 'COUT', bel.bel, 'CO3',
-                                [('site_pip', 'COUTUSED', '0')])
+
             site.add_internal_source(bel, co_pin, lut + '_CY', bel.bel,
                                      'CO{}'.format(idx))
+            if idx == 3:
+                # Connects internal pin CO[3] to site pin COUT
+                site.add_output_from_internal('COUT', lut + '_CY')
```

Comment:
I have bumped into an issue that can get solved by this PR, but this is missing an important piece actually, which is the fact that the COUT wire needs to pass through the COUTUSED site pip. If this is not added, than there will be an unrouted net when generating the physical netlist, and a resulting invalid DCP.

```suggestion
                site.add_output_from_internal('COUT', lut + '_CY', [('site_pip', 'COUTUSED', '0')])
```